### PR TITLE
#162949029 Setting up redux

### PR DESCRIPTION
**Pull request by waracci**
#### What does this PR do?

This PR implements react-redux using the `redux-saga` library.

#### Description of Task to be completed?

- [x] Implement Redux store
- [x] set up necessary files needed to create the store

#### How should this be manually tested?

1. Clone this repository on your local machine using 

     `git clone https://github.com/andela/ah-robotics-frontend.git`
2. Change your working directory to the project's directory using 
     
     `cd ah-robotics-frontend`
3. Change to the specific branch `ch-setup-redux-162949029` using 

     `git checkout ch-setup-redux-162949029`

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm start` command to start the application.

6. The application should start successfully displaying a page as shown below.

screenshot 2019-02-07 at 12 39 52

7. Click on the `add-button` it should increment the count as displayed above.

8. Click on the `Redux DevTools` extension on the Chrome browser and it should display the store as shown below.

> The store will update and it will display the count number

> Ps. If you **do not** have `Redux DevTools` installed, find the instructions listed below on how to install the extension.

screenshot 2019-02-06 at 18 53 29

#### Additional steps to set up your browser for testing React Apps better.

> These tools are not necessary to run the application, they are just extra tools to help you test and debug React applications quickly

1. Install React DevTools extension for Chrome

> Find the extension [Here](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi)

2. Install Redux extension for Chrome

>Find the extension [Here](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en)

#### What are the relevant pivotal tracker stories?

[#162949029](https://www.pivotaltracker.com/story/show/162949029)

### CODE REVIEW

**Comment by Teatoller** [comment]
Hello @waracci 
I have followed the steps successfully.
The only difference I encounter is that  under state I have ```(pin): undefined``` . Whereas you have ```app (pin): {}```.
What could be be the reason?


screenshot 2019-02-07 at 06 11 26

Regrads
Steven Ennis

**Comment by Teatoller** [comment]
> Hello @waracci
> I have followed the steps successfully.
> The only difference I encounter is that under state I have `(pin): undefined` . Whereas you have `app (pin): {}`.
> What could be be the reason?
> 
> screenshot 2019-02-07 at 06 11 26
> 
> Regrads
> Steven Ennis
Hey it has worked.
It is okay

**Comment by waracci** [comment]
Hey @Teatoller, I am glad it worked on your end. Thanks for your time!

File: src/store/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { createStore, applyMiddleware } from "redux";
+import createSagaMiddleware from "redux-saga";
+import rootreducer from "./rootReducer";
+import { composeWithDevTools } from "redux-devtools-extension";
+
+import rootSaga from "../sagas";
+
+const configureStore = () => {
```
Comment (jomasim) [inline src/store/index.js:8]: Add descriptive comments to this function.

Comment (waracci) [reply src/store/index.js:8]: Okay, let me look into this.

File: src/containers/app/App.test.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App';
+import App from '../../App';
 
 it('renders without crashing', () => {
```
Comment (sharkdevs) [inline src/containers/app/App.test.js]: I see this is a test. However, You have not included steps to run it in the PR. You can consider that.

**Review by sharkdevs — COMMENTED** [review]
All the steps in the PR description are working fine. Consider the comment below:

**Review by johnwayodi — COMMENTED** [review]
@waracci , I managed to follow the instructions and obtained the same result, nice work.

File: src/containers/app/App.test.js (hunk 2 above)
Comment (waracci) [reply src/containers/app/App.test.js]: I was setting up redux, @joma is supposed to ensure the tests are configured correctly.

**Comment by Teatoller** [comment]
Hello @waracci 
I have test the additional features ```7``` and ```8``` and they are working as expected.
>Click on the `add-button` it should increment the count as displayed above.
Click on the `Redux DevTools` extension on the Chrome browser and it should display the store as shown below.

Regards
Steven Ennis

**Review by Paulvitalis200 — COMMENTED** [review]
Hey @waracci ,
I followed the steps and managed to test this chore on my machine. It works well. However, I had to restart the server for the count to be shown in the redux developer tools. At first, it was showing **undefined** You can look into that or probably add a step in the Pull request to handle cases like that. Otherwise good job on implementing the feature. It works.

File: src/containers/app/App.test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<<<<<<< HEAD
+import React from 'react';
+<<<<<<< HEAD:src/App.test.js
+import { shallow } from 'enzyme';
+import App from './App';
+=======
```
Comment (Collins33) [inline src/containers/app/App.test.js]: I think you need to check on this merge conflict statements that are causing tests to fail

**Review by Collins33 — CHANGES_REQUESTED** [review]

File: src/containers/app/App.test.js (hunk 3 above)
Comment (waracci) [reply src/containers/app/App.test.js]: Thanks for pointing this out, am working on it.

**Pull request merged**